goaviatrix/firenet.go

```
@@ -0,0 +1,251 @@
+		return errors.New(("failed to read interface information: ") + err.Error())
+	}
+
+	associateFirewallWithFireNet := url.Values{}
+	associateFirewallWithFireNet.Add("CID", c.CID)
+	associateFirewallWithFireNet.Add("action", "associate_firewall_with_firenet")
+	associateFirewallWithFireNet.Add("vpc_id", firewallInstance.VpcID)
+	associateFirewallWithFireNet.Add("gateway_name", firewallInstance.GwName)
+	associateFirewallWithFireNet.Add("firewall_id", firewallInstance.InstanceID)
+	associateFirewallWithFireNet.Add("firewall_name", firewallInstance.FirewallName)
+	associateFirewallWithFireNet.Add("lan_interface", interfaces[0])
+	associateFirewallWithFireNet.Add("egress_interface", interfaces[1])
+	associateFirewallWithFireNet.Add("management_interface", interfaces[2])
+
+	Url.RawQuery = associateFirewallWithFireNet.Encode()
+	resp, err := c.Get(Url.String(), nil)
+	if err != nil {
+		return errors.New("HTTP Get associate_firewall_with_firenet failed: " + err.Error())
+	}
+	var data APIResp
+	if err = json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return errors.New("Json Decode associate_firewall_with_firenet failed: " + err.Error())
+	}
+	if !data.Return {
+		return errors.New("Rest API associate_firewall_with_firenet Get failed: " + data.Reason)
+	}
+	return nil
+}
+
+func (c *Client) DisassociateFirewallFromFireNet(firewallInstance *FirewallInstance) error {
+	Url, err := url.Parse(c.baseURL)
+	if err != nil {
+		return errors.New(("url Parsing failed for disassociate_firewall_with_firenet: ") + err.Error())
+	}
+	disassociateFirewallWithFireNet := url.Values{}
+	disassociateFirewallWithFireNet.Add("CID", c.CID)
+	disassociateFirewallWithFireNet.Add("action", "disassociate_firewall_with_firenet")
+	disassociateFirewallWithFireNet.Add("vpc_id", firewallInstance.VpcID)
+	disassociateFirewallWithFireNet.Add("firewall_id", firewallInstance.InstanceID)
+
+	Url.RawQuery = disassociateFirewallWithFireNet.Encode()
+	resp, err := c.Get(Url.String(), nil)
```

Might need to ignore error if it is already disassociated